.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+repos:
+-   repo: https://github.com/psf/black
+    rev: 23.7.0
+    hooks:
+    - id: black
+      args: [--line-length=100]
+      exclude: ^(bin/|doc/|scripts/)
+- repo: local
+  hooks:
+    - id: pylint
+      name: pylint
+      entry: pylint
+      language: system
+      types: [python]
+      args:
+        [
+          "-rn", # Only display messages
+          "-sn", # Don't display the score
+          "--rcfile=.pylintrc", # Link to your config file
+        ]
+      exclude: ^(bin/|doc/|scripts/|tests/)
```

Comment:
This will force us to install black before use, or it installs it and then runs automatically?

Reply:
Seems to be getting `black` and `pylint` somehow, because I tested in Perlmutter in a clean `venv` and it works.

Reply:
May be worth noting for everybody (core included) later if using this. It could potentially (if not already) be rolled out across the other repos.